src/main/java/org/gridsuite/shortcircuit/server/repositories/specifications/SpecificationUtils.java

```diff
@@ -116,21 +131,41 @@ private static <X> Specification<X> appendTextFilterToSpecification(Specificatio
 
     @NotNull
     private static <X> Specification<X> appendNumberFilterToSpecification(Specification<X> specification, ResourceFilter resourceFilter) {
-        Specification<X> completedSpecification = specification;
+        Specification<X> completedSpecification;
+        String value = resourceFilter.value().toString();
 
-        // We need to cast it as String before and use .valueOf to be able to works with integers
-        Double value = Double.valueOf(resourceFilter.value().toString());
+        if (isInteger(value)) {
```

Comment:
it looks weird, if user writes "2", we don't use epsilon, if he writes "2.0", we do use it ?

Reply:
Normally, when writing values without a comma, it will be on integer filters. However, I can leave the epsilon even on integer values

Reply:
I think it's something we should discuss with PO, how is the filter supposed to work ?